### ISSUE TITLE
Where to find vtk.jar?

### ISSUE THREAD

**Issue by arianaa30** [j-mp/JPLYVIS#1]
Hi,
I was trying to run your project, but it needs a library called vtk.jar before I run. I couldn't find it online. Can you share it please?
I can not run the project without it.

**Comment by j-mp** [comment]
Hi,

sorry I missed to add it, but anyway it is not essential. It just belongs to the a vtk demo. Therefore I compiled the vtk source with java bindings.

Okay, to run the project, just remove the vtk.jar from the java build path. There will be some errors just ignore it or remove the code which belongs to the vtk demo.

Bests,

JMP
________________________________
Von: arianaa30 [notifications@github.com]
Gesendet: Montag, 26. Juni 2017 00:40
An: j-mp/JPLYVIS
Cc: Subscribed
Betreff: [j-mp/JPLYVIS] Where to find vtk.jar? (#1)


Hi,
I was trying to run your project, but it needs a library called vtk.jar before I run. I couldn't find it online. Can you share it please?
I can not run the project without it.

**Comment by arianaa30** [comment]
I was trying to remove the unused code. These classes are: `DemoJavaVTK.java, PLYVISMenuBar.java, and MeshTools.java` . After removing them, which file should I run? if it is the `main`, it runs, but nothing happens.

**Comment by arianaa30** [comment]
Actually it seems like it runs after a long time wait. It is very slow even on the application startup (prior to loading `ply `file). Was it slow on your end as well? Or I am missing something?

**Comment by j-mp** [comment]
Hi,

the GUI should load without delay and also the loading of a file should be okay. Per default the mesh triangulation is used, the works quite fast, but if you use the primitive visualization it is really slow. What kind of data you are using and what you are going to do? I designed the application to open some proprietary ply files and include this into an existing java framework. What kind of system you use?

Greetings, JMP
________________________________________
Von: arianaa30 [notifications@github.com]
Gesendet: Montag, 26. Juni 2017 22:19
An: j-mp/JPLYVIS
Cc: Jean-Michel Pape; Comment
Betreff: Re: [j-mp/JPLYVIS] Where to find vtk.jar? (#1)

Actually it seems like it runs after a long time wait. It is very slow loading a few MB ply file. Was it slow on your end as well? Or I am missing something?

**Comment by arianaa30** [comment]
I was trying to find a PC visualization tool in Java, maybe add network streaming to it. Yours seemed good. I restarted the system and now the load is fast. One problem is that it only shows points coordinates, not the colors (they are not even parsed). All points are green now. I would like to show the colors as well.

**Comment by arianaa30** [comment]
I was trying to add the feature of colored PCs. I am able to show colored primitives, but for Mesh it wasn't easy. So in primitive you draw points individually while in the other one you create a mesh?

**Comment by j-mp** [comment]
Yes, you can only visualize color by using primitives, for some reason (I used it to visualize plant scans) it will only show some green color, but it is easy to change this (you tried already?).
When I wrote the code it wasn´t possible to add colors to a triangle mesh in javafx. So yes, by using the primitive view I draw for every point a primitive what will become very slow if you have a lot of points, and yes for the other view I create a triangle mesh.

Anyway if you want to contribute, I can add you to the project ...

Greetings, JMP
________________________________
Von: arianaa30 [notifications@github.com]
Gesendet: Dienstag, 27. Juni 2017 18:38
An: j-mp/JPLYVIS
Cc: Jean-Michel Pape; Comment
Betreff: Re: [j-mp/JPLYVIS] Where to find vtk.jar? (#1)


I was trying to add the feature of colored PCs. I am able to show colored primitives, but for Mesh it wasn't easy. So in primitive you draw points individually while in the other one you create a mesh?

**Comment by arianaa30** [comment]
Yes, I added the color for the primitive. For some reason it doesn't look like the original color though! Bad that it is slow.
I am currently in experimental phase but if I ended up with Java I will for sure consider this and will contribute.

**Comment by j-mp** [comment]
Hi,

could be that there are some effects enabled (Lightsource color, etc.) which influences the object colors. It maybe possible to use the triangle mesh with different colors, but you have to create a new one for each color. I dont know how this would be perform ...
________________________________
Von: arianaa30 [notifications@github.com]
Gesendet: Mittwoch, 28. Juni 2017 16:19
An: j-mp/JPLYVIS
Cc: Jean-Michel Pape; Comment
Betreff: Re: [j-mp/JPLYVIS] Where to find vtk.jar? (#1)


Yes, I added the color for the primitive. For some reason it doesn't look like the original color though! Bad that it is slow.
I am currently in experimental phase but if I ended up with Java I will for sure consider this and will contribute.
